IGAPP-934: Poi list view for desktop

f1sh1918 opened the pull request:
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

http://localhost:9000/testumgebung/de/locations

f1sh1918 commented on web/src/components/FeedbackModal.tsx:
```diff
@@ -27,7 +27,7 @@ const ModalContainer = styled.div`
   bottom: 0;
   left: 0;
-  z-index: 11;
+  z-index: 111;
```
Modal should always be on top. in this case over the `BottomActionSheet`

SteffiStuffel replied:
111 is not a good value though.

If it needs to be bigger than 110, use 120, so we still have the possibility to put a element between, if needed in the future. If 100 works, use this.

f1sh1918 commented on web/src/components/Footer.tsx:
```diff
@@ -13,6 +13,7 @@ const FooterContainer = styled.footer`
   justify-content: center;
   padding: 15px 5px;
   background-color: ${props => props.theme.colors.backgroundAccentColor};
+  box-shadow: 0 2px 3px 3px rgba(0, 0, 0, 0.1);
```
added box shadow to look similar to the `Header` box shadow and to separate from the PoiList Toolbar

SteffiStuffel replied:
Looking at the designs the footer should not look like this on the maps page, but I guess it is not part of this ticket.

f1sh1918 replied:
yes we have to decide what happens with the footer, maybe it gonna be a small overlay on the map view. But yes separate ticket

f1sh1918 commented on web/src/components/Layout.tsx:
```diff
@@ -28,7 +28,7 @@ const RichLayout = styled.div`
 `
 
-const Body = styled.div`
+const Body = styled.div<{ fullWidth: boolean }>`
```
added to make layout fullwidth for Pois Page

f1sh1918 commented on web/src/components/Toolbar.tsx:
```diff
@@ -36,18 +35,28 @@ const Headline = styled.h5`
-    {children}
-  </ToolbarContainer>
-)
+const Toolbar: React.FC<ToolbarProps> = ({
```
refactored to functional component

klinzo replied:
Was a functional component before already? :D

f1sh1918 replied:
oh sorry your are right :D just removed HOC

f1sh1918 commented on web/src/constants/dimensions.ts:
```diff
@@ -14,8 +17,11 @@ const dimensions: DimensionsType = {
-  headerHeightSmall: 70
+  headerHeightSmall: 70,
+  footerHeight: 50,
+  navigationMenuHeight: 90
```
heights are needed to calculate the `main`container for the pois page

SteffiStuffel commented on web/src/components/LocationToolbar.tsx:
```diff
@@ -7,14 +7,15 @@ type PropsType = {
 class LocationToolbar extends React.PureComponent<PropsType> {
   render(): ReactNode {
-    const { viewportSmall, children, openFeedbackModal } = this.props
+    const { viewportSmall, children, openFeedbackModal, iconDirection = 'column' } = this.props
```
no need to set the default value here, as it is set to "column" in `Toolbar.tsx` anyway if not set. I think it is better not to set the default twice.

SteffiStuffel commented on web/src/components/PoisDesktop.tsx:
```diff
@@ -0,0 +1,66 @@
+import dimensions from '../constants/dimensions'
+
+const ListViewWrapper = styled.div<{ panelHeights: number }>`
+  min-width: 370px;
```
I think 370 is to big. 150 is also enough otherwise if the display is very small, but still counts as a desktop, there is almost no card visiable but only the sidebar.

f1sh1918 replied:
there are always edge cases in responsive design. Here is now a pic of 770px. Since we have no tablet design, we can either start showing desktop mode for large desktops `>992` and use the bottom action bar instead before. but i can not decrease the size for the panel even more

![image](https://user-images.githubusercontent.com/37902063/156561204-019252ee-c119-455d-a1e4-f84f33dc7557.png)

SteffiStuffel replied:
Not really a edge case. I just think it would be better / more responsive design if the sidebar width is more responsive. Any argument against changing the min-width to a smaller value?

SteffiStuffel commented on web/src/components/PoisDesktop.tsx:
```diff
@@ -0,0 +1,66 @@
+  padding-top: 32px;
+  padding-bottom: 20px;
+  text-align: center;
+  font-size: 18px;
```
somehow frustrating there is no appropriate fontSize in the theme for this.

f1sh1918 replied:
yea we have to refactor some typo stuff definitly but even think if we want to use css clamp for better responsiveness
https://developer.mozilla.org/en-US/docs/Web/CSS/clamp()

SteffiStuffel commented on web/src/hooks/useWindowDimensions.ts:
```diff
@@ -1,6 +1,6 @@
 import { useState, useEffect } from 'react'
 
-const VIEWPORT_SMALL_THRESHOLD = 750
+const VIEWPORT_SMALL_THRESHOLD = 768
```
Why is this such a weird number?

f1sh1918 replied:
this is a standard viewport width used in many frameworks. 750 is weird ;) f.e. here under medium
https://getbootstrap.com/docs/4.0/layout/grid/#grid-options

SteffiStuffel reviewed (commented):
Tested on Chrome:
- Scrollbar looks shitty for me, but I think it is just the standart scrollbar of the browser, but in the "middle" of the screen it seems to big/clumsy.
- "Haben Sie Feedback" text is missing, it is there for mobile view and also in the designs. Can't the component rendering the feedback be shared so we always have consitent behaviour?
- Some details from the design are not implementet correctly, e.g. the separation lines are to thick / dark.
- distance is not displayed for me, but I also did not allow location permission (as i wasn't asked), so this may be fine
- For better responsiveness it would be nice it the left part would be a little responsive (scale down, so the layout for the smalles-desktop-devices still looks good)

Can you also add few tests?

klinzo commented:
> Tested on Chrome:
> 
>     * Scrollbar looks shitty for me, but I think it is just the standart scrollbar of the browser, but in the "middle" of the screen it seems to big/clumsy.
> 
>     * "Haben Sie Feedback" text is missing, it is there for mobile view and also in the designs. Can't the component rendering the feedback be shared so we always have consitent behaviour?
> 
>     * Some details from the design are not implementet correctly, e.g. the separation lines are to thick / dark.
> 
>     * distance is not displayed for me, but I also did not allow location permission (as i wasn't asked), so this may be fine
> 
Probably you need to start using `yarn start --https`

klinzo commented on web/src/routes/PoisPage.tsx:
```diff
@@ -154,32 +171,52 @@ const PoisPage = ({ cityCode, languageCode, cityModel, pathname, languages }: Ci
+            poiList={poiList}
+          />
         )}
-        {/* TODO add feedback toolbar IGAPP-914 */}
```
Next time better separate the changes and create two PRs.

klinzo commented on web/src/components/PoisDesktop.tsx:
```diff
@@ -0,0 +1,66 @@
+      <div>
+        <ListViewWrapper panelHeights={panelHeights}>
+          <ListHeader>{currentFeature?.properties.title || t('listTitle')}</ListHeader>
+          {!currentFeature && sortedPois.length > 0 && poiList}
```
I think it is quite weird that you pass `currentFeature` and `sortedPois` only to decide whether to render the list or not.
The check for the amount of sorted pois is definitely not needed as we probably want the message "no pois" to show:
```suggestion
          {!currentFeature && poiList}
```
The check for `currentFeature` can probably be removed once you remove the details view/move it to the sidebar, right?

f1sh1918 replied:
no, because we then pass here the details view and render it
`{currentFeature ? poiDetails : poiList}`

klinzo commented on web/src/routes/PoisPage.tsx:
```diff
@@ -154,32 +171,52 @@ const PoisPage = ({ cityCode, languageCode, cityModel, pathname, languages }: Ci
+        {viewportSmall ? (
+          <PoisMobile
+            currentFeature={currentFeature}
+            toolbar={toolbar(setFeedbackModalRating)}
```
```suggestion
            toolbar={toolbar}
```

klinzo commented on web/src/routes/PoisPage.tsx:
```diff
@@ -154,32 +171,52 @@ const PoisPage = ({ cityCode, languageCode, cityModel, pathname, languages }: Ci
+  )
+
+  const poiList = <List noItemsMessage={t('noPois')} items={sortedPois} renderItem={renderPoiListItem} borderless />
+  // To get the map fullheight its reduced by header, footer, navMenu
```
Personal opinion: I think this comment does not really add any necessary information

f1sh1918 replied:
check last commit

f1sh1918 commented on web/src/components/PoisMobile.tsx:
```diff
@@ -0,0 +1,34 @@
+      <>
+        {mapView}
+        <BottomActionSheet title={currentFeature?.properties.title || t('listTitle')} toolbar={toolbar} ref={ref}>
+          {sortedPois.length > 0 && !currentFeature && poiList}
```
check comment above

f1sh1918 commented:
> Tested on Chrome:

> * "Haben Sie Feedback" text is missing, it is there for mobile view and also in the designs. Can't the component rendering the feedback be shared so we always have consitent behaviour?

It gonna be changed in the designs, since its never displayed it on desktop view
> 
> * Scrollbar looks shitty for me, but I think it is just the standart scrollbar of the browser, but in the "middle" of the screen it seems to big/clumsy.

This quite difficult since there are `scrollbar-kits` but not supported by all browsers like firefox. If we have fixed footer container i can not avoid this
> * Some details from the design are not implementet correctly, e.g. the separation lines are to thick / dark.
The used colors is not in design, gonna talk to nikola

> * distance is not displayed for me, but I also did not allow location permission (as i wasn't asked), so this may be fine
Hm should be browser preferences it def. works even you have to `yarn start --https`
Even the distance calc gonna be fixed in another pr, so value is atm wrong

> * For better responsiveness it would be nice it the left part would be a little responsive (scale down, so the layout for the smalles-desktop-devices still looks good)
> 
I added clamp() for responsive padding, fonts, images, check it out
> Can you also add few tests?
I wasn't sure what to test here because there are not real "new" components with logic, so if there are any suggestions. But i also gonna think about it once more!

f1sh1918 commented on web/src/components/LocationLayout.tsx:
```diff
@@ -46,6 +46,7 @@ const LocationLayout = (props: PropsType): ReactElement => {
 
   return (
     <Layout
+      fullWidth={route === POIS_ROUTE}
```
check last commit

klinzo reviewed (approved):
More things I encountered:
* No pois are shown until location request has been accepted or denied. This makes the layout/ui jump since the pois are only shown afterwards and the sidebar is smaller if only a nothing found message is displayed.
* Navigating back/clicking on `Karte` after selecting on a poi does nothing (except changing the url)

The design/responsiveness is really well done though.

f1sh1918 commented:
> 

@klinzo which browser did you test?
I tested on chrome, firefox and mobile phone and deselection works for me (clicking on the map)

klinzo commented:
> > 
> 
> @klinzo which browser did you test? I tested on chrome, firefox and mobile phone and deselection works for me (clicking on the map)

I meant clicking on the header navigation item "Karte", sorry. Clicking on the map only works if there is no poi in the press area. If you click on another poi, nothing happens.

f1sh1918 commented:
> > > 
> > 
> > 
> > @klinzo which browser did you test? I tested on chrome, firefox and mobile phone and deselection works for me (clicking on the map)
> 
> I meant clicking on the header navigation item "Karte", sorry. Clicking on the map only works if there is no poi in the press area. If you click on another poi, nothing happens.

ok pls create a bug ticket for "Karte" if you click on another poi this poi gets selected so not sure what else should happen
